openedx/core/djangoapps/courseware_api/tests/test_views.py

```diff
@@ -303,10 +303,24 @@ def test_streak_data_in_response(self):
         CourseEnrollment.enroll(self.user, self.course.id, 'audit')
         with override_waffle_flag(COURSEWARE_MFE_MILESTONES_STREAK_DISCOUNT, active=True):
             with mock.patch('common.djangoapps.student.models.UserCelebration.perform_streak_updates', return_value=3):
-                response = self.client.get(self.url, content_type='application/json')
-                celebrations = response.json()['celebrations']
-                assert celebrations['streak_length_to_celebrate'] == 3
-                assert celebrations['streak_discount_enabled'] is True
+                with mock.patch('common.djangoapps.track.segment.track') as mock_segment_track:
+                    response = self.client.get(self.url, content_type='application/json')
+                    celebrations = response.json()['celebrations']
+                    assert celebrations['streak_length_to_celebrate'] == 3
+                    assert celebrations['streak_discount_enabled'] is True
+                    mock_segment_track.assert_called_once()
+
+    def test_streak_segment_suppressed_for_unverified(self):
+        """ Test that metadata endpoint does not return a discount for unverified and signal is not sent """
```

Comment:
i think the comment here needs to be adjusted to something like
does not return a discount when the flag is disabled
(you're not setting verified here)

Reply:
Thanks. Copy and pasta error. Fixed